### PR TITLE
Bolt protocol

### PR DESCRIPTION
**Pull request by AviAvni**

### CODE REVIEW

File: src/bolt/bolt.h [hunk 1]
```diff
@@ -0,0 +1,258 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (AviAvni) [reply src/bolt/bolt.h:6]: no I wrote everything

File: src/bolt/bolt.h [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    uint32_t size
+);
+
+void bolt_reply_map
+(
+    bolt_client_t *client,
+    uint32_t size
+);
+
+void bolt_reply_structure
```
[the remark this answers is not in the document]
Comment (AviAvni) [reply src/bolt/bolt.h:171]: yes

File: src/bolt/bolt.h [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+bool bolt_check_handshake
+(
+    socket_t socket
+);
+
+bolt_version_t bolt_read_supported_version
+(
+    socket_t socket
+);
```
[the remark this answers is not in the document]
Comment (AviAvni) [reply src/bolt/bolt.h:275]: there is one you always add 2 lines

File: src/bolt/socket.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    socket_t socket,
+    char* buff,
+    size_t size
+);
+
+#define socket_accept(socket) accept(socket, NULL, NULL)
+
+#define socket_write(socket, buff, size) send(socket, buff, size, 0)
+
+#define socket_close(socket) close(socket)
```
[the remark this answers is not in the document]
Comment (AviAvni) [reply src/bolt/socket.h]: yes easier to know that we use socket

File: src/module.c [hunk 5]
```diff
@@ -93,6 +96,253 @@ static int GraphBLAS_Init(RedisModuleCtx *ctx) {
...
+) {
+	bolt_client_t *client = (bolt_client_t*)user_data;
+	if(!socket_read(fd, client->read_buffer + client->read_index, 2)) {
+		rm_free(client);
+		socket_close(fd);
+		RedisModule_EventLoopDel(fd, REDISMODULE_EVENTLOOP_READABLE);
+		return;
+	}
+
+	uint16_t size = bswap_16(*(uint16_t*)(client->read_buffer + client->read_index));
```
[the remark this answers is not in the document]
Comment (AviAvni) [reply src/module.c:252]: bolt is using big-ending format

File: src/module.c [hunk 6]
```diff
@@ -93,6 +96,253 @@ static int GraphBLAS_Init(RedisModuleCtx *ctx) {
...
+
+	if(size > 0) {
```
[the remark this answers is not in the document]
Comment (AviAvni) [reply src/module.c:254]: yes bolt have a message named NOOP which is size 0

File: tests/tck/features/clauses/create/Create6.feature
Comment (postpersonality) [inline tests/tck/features/clauses/create/Create6.feature:1]: Do you run these feature-tests both for Bolt and Redis protocols?

**Pull request merged**